dotCMS/src/main/java/com/dotcms/rendering/velocity/directive/DotParse.java

```diff
@@ -1,159 +1,239 @@
-        throw e;
-    } catch (DotSecurityException  e) {
-        Logger.warn(this.getClass(), " - unable to resolve " + templatePath + " getting this: "+ e.getMessage() );
-        if(e.getStackTrace().length>0){
-            Logger.warn(this.getClass(), " - at " + e.getStackTrace()[0]);
+
+        tokens.nextToken();
+        final String inodeOrIdentifier = tokens.nextToken();
+        final String fieldVar = tokens.hasMoreTokens() ? tokens.nextToken() : DotAssetContentType.ASSET_FIELD_VAR;
+        final Optional<ShortyId> shortOpt = APILocator.getShortyAPI().getShorty(inodeOrIdentifier);
+
+        if (!shortOpt.isPresent()) {
+            return Optional.empty();
         }
-        throw new ResourceNotFoundException(e);
-    } catch (Exception e) {
-        Logger.warn(this.getClass(), " - unable to resolve " + templatePath + " getting this: "+ e.getMessage() );
-        if(e.getStackTrace().length>0){
-            Logger.warn(this.getClass(), " - at " + e.getStackTrace()[0]);
+
+        ShortyId shorty = shortOpt.get();
+        final Optional<Contentlet> conOpt = (shorty.type == ShortType.IDENTIFIER)
+                        ? APILocator.getContentletAPI().findContentletByIdentifierOrFallback(shorty.longId, params.mode.showLive,
+                                        params.language.getId(), params.user, false)
+                        : Optional.ofNullable(
+                                        APILocator.getContentletAPI().find(shorty.longId, params.user, params.mode.respectAnonPerms));
+
+        if (!conOpt.isPresent()) {
+            return Optional.empty();
         }
-        throw new DotStateException(e);
+        
+        Identifier identifier = APILocator.getIdentifierAPI().find(conOpt.get().getIdentifier());
+        return Optional.of(Tuple.of(identifier, fieldVar));
+
+
     }
-  }
 
-    private void throwNotResourceNotFoundException(final RenderParams params, final String templatePath) {
-        final String errorMessage = String.format("Not found %s version of [%s]", params.mode, templatePath);
-        throw new ResourceNotFoundException(errorMessage);
+    private Optional<Tuple2<Identifier, String>> resolveFileAsset(final Context context, final RenderParams params,
+                    String templatePath) throws DotDataException, DotSecurityException {
+        final User user = params.user;
+        Host host = params.currentHost;
+        // if we have a host
+        if (templatePath.startsWith(hostIndicator)) {
+            templatePath = templatePath.substring(hostIndicator.length(), templatePath.length());
+            String hostName = templatePath.substring(0, templatePath.indexOf('/'));
+            templatePath = templatePath.substring(templatePath.indexOf('/'), templatePath.length());
+            host = APILocator.getHostAPI().resolveHostName(hostName, user, params.mode.respectAnonPerms);
+        }
+
+
+        Identifier identifier = APILocator.getIdentifierAPI().find(host, templatePath);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'identifier' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5627236)